File: blue_ticker/utils/master_types.py
"""
銘柄マスタ関連の TypedDict 定義

CSV 読み込み後の内部表現と、検索/業種一覧の公開結果を型で表現する。
"""

from typing import TypedDict


class MasterStock(TypedDict):
    Code: str               # 証券コード（4桁または5桁）。未上場は空文字
    EdinetCode: str         # ＥＤＩＮＥＴコード（E00004 形式）
    CoName: str             # 提出者名
    CoNameEn: str           # 提出者名（英字）
    CoNameKana: str         # 提出者名（ヨミ）
    CoNameNormalized: str   # 検索用正規化名称
    S33Nm: str              # 提出者業種
    MktNm: str              # 上場区分（"上場" / "非上場"）
    IsListed: bool          # 上場かどうか
    HasConsolidated: bool   # 連結の有無


class StockSearchResult(TypedDict):
    code: str
    name: str
    sector: str
    market: str


class SectorSummary(TypedDict):
    code: str
    name: str
    count: int

File: Sources/BlueTicker/Models/MasterTypes.swift
struct MasterStock: Codable {
    let code: String
    let coName: String
    let coNameUpper: String
    let coNameNormalized: String
    let s33nm: String
    let mktNm: String
    let location: String
    /// EDINET「提出者業種」の値。CSV に業種コード列が無いため、現状は業種名と同じ文字列が入る。
    let s33: String
    /// EDINET「提出者種別」の値（例: "内国法人・組合" / "外国法人・組合"）。
    let filerType: String

    // CSV の列名と対応させる CodingKeys
    enum CodingKeys: String, CodingKey {
        case code      = "Code"
        case coName    = "CoName"
        case coNameUpper       = "CoNameUpper"
        case coNameNormalized  = "CoNameNormalized"
        case s33nm     = "S33Nm"
        case mktNm     = "MktNm"
        case location  = "Location"
        case s33       = "S33"
        case filerType = "FilerType"
    }
}

struct StockSearchResult: Codable {
    let code: String
    let name: String
    let sector: String
    let market: String
    let location: String
}

struct SectorSummary: Codable {
    let code: String
    let name: String
    let count: Int
}
